common/app/common/package.scala

```diff
@@ -43,39 +43,59 @@ object `package`
-      throw error
-    case timeout: DCRTimeoutException =>
-      log.error(s"Got a timeout while calling DCR: ${timeout.getMessage}, path: ${request.path}", timeout)
-      NoCache(GatewayTimeout(timeout.getMessage))
-
-    case timeout: TimeoutException =>
-      log.error(s"Got a timeout while calling content api: ${timeout.getMessage}, path: ${request.path}", timeout)
-      NoCache(GatewayTimeout(timeout.getMessage))
-
-    case error =>
-      log.error(s"Content api exception: ${error.getMessage}", error)
-      Option(error.getCause).foreach { cause =>
-        log.error(s"Content api exception cause (path: ${request.path}: ", cause)
-      }
-      NoCache(InternalServerError)
+    val fastlyRequestId = request.headers.get("x-gu-xid").getOrElse("fastly-id-not-provided")
+    val customFieldMarker: LogstashMarker = {
+      append("fastlyRequestId", fastlyRequestId)
+    }
+    {
+      case _: CircuitBreakerOpenException =>
+        log.logger.error(
+          customFieldMarker,
+          s"Got a circuit breaker open error while calling content api, path: ${request.path}",
+        )
+        NoCache(ServiceUnavailable)
+      case ContentApiError(404, message, _) =>
+        log.logger.info(customFieldMarker, s"Got a 404 while calling content api: $message, path:${request.path}")
+        NoCache(NotFound)
+      case ContentApiError(410, message, errorResponse) =>
+        errorResponse match {
+          case Some(errorResponse) if isCommercialExpiry(errorResponse) =>
+            log.logger.info(
+              customFieldMarker,
+              s"Got a 410 while calling content api: $message: ${errorResponse.message}, path: ${request.path}",
+            )
+            Cached(60)(RevalidatableResult.Ok(views.html.commercialExpired()))
+          case _ =>
+            log.info(s"Got a 410 while calling content api: $message, path: ${request.path}")
+            NoCache(Gone)
+        }
+
+      // Custom DCR exceptions to distinguish from CAPI/other backend errors.
+      case error: DCRLocalConnectException =>
+        throw error
+      case timeout: DCRTimeoutException =>
+        log.logger.error(
+          customFieldMarker,
+          s"Got a timeout while calling DCR: ${timeout.getMessage}, path: ${request.path}",
+          timeout,
+        )
+        NoCache(GatewayTimeout(timeout.getMessage))
+
+      case timeout: TimeoutException =>
+        log.logger.error(
+          customFieldMarker,
+          s"Got a timeout while calling content api: ${timeout.getMessage}, path: ${request.path}",
+          timeout,
+        )
+        NoCache(GatewayTimeout(timeout.getMessage))
+
+      case error =>
+        log.logger.error(customFieldMarker, s"Content api exception: ${error.getMessage}", error)
+        Option(error.getCause).foreach { cause =>
+          log.logger.error(customFieldMarker, s"Content api exception cause (path: ${request.path}: ", cause)
+        }
+        NoCache(InternalServerError)
+    }
```

Comment:
This function `convertApiExceptionsWithoutEither` is handling the api call errors. It's used with `recover` on Futures that are calling the content api, hence, recover catches exceptions that occur during the execution of the Future and provides a fallback value (or transformation) for these errors.

The logs on this function were not including any custom field, so by adding a fieldMarket `append("fastlyRequestId", fastlyRequestId)` the fastlyRequestId is added as a custom field to these logs.